app/src/main/java/com/example/erabook/data/firebasedb/UserDataRemote.kt

```diff
@@ -21,6 +21,6 @@ data class Coordinates(
 )
 
 data class FavoriteBooks(
-    @SerializedName("book_title") var bookTitle: String,
-    @SerializedName("book_author") var bookAuthor: String,
+    @SerializedName("book_title") var bookTitle: String? = null,
+    @SerializedName("book_author") var bookAuthor: String? = null,
```

Comment:
Why you have set all the values as "var" is there any specific reason?
This may introduce thread not safety issue. Suppose you are trying to access this value and from another class you change this value since it's var. An on that case you would not have the value that you want.
Please make them all as "val" variables.